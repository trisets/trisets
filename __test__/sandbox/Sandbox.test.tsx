import Sandbox from '@/pages/sandbox';
import { cleanup, fireEvent, render, screen, waitFor } from '@testing-library/react';
import { afterEach, expect, test } from 'vitest';

afterEach(() => cleanup());

test('/sandbox snapshot test', async () => {
  const { container } = render(<Sandbox />);
  expect(container.firstChild).toMatchSnapshot();
});

test('push get-user button', async () => {
  render(<Sandbox />);
  fireEvent.click(screen.getByTestId('get-user'));
  await waitFor(
    () => {
      expect(screen.getByText('ユーザー情報取得失敗！(サインインしてないかも)')).toBeDefined();
    },
    { timeout: 1000 }
  );
});

test('push sign-in button', async () => {
  render(<Sandbox />);
  fireEvent.click(screen.getByTestId('sign-in'));
  await waitFor(
    () => {
      expect(screen.getByText('サインイン成功！')).toBeDefined();
    },
    { timeout: 1000 }
  );
});

test('push get-user button', async () => {
  render(<Sandbox />);
  fireEvent.click(screen.getByTestId('get-user'));
  await waitFor(
    () => {
      expect(screen.getByText('ユーザー情報取得成功！')).toBeDefined();
    },
    { timeout: 1000 }
  );
});
