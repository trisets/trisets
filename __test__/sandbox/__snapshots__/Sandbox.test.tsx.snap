// Vitest Snapshot v1

exports[`/sandbox snapshot test 1`] = `
<div>
  <div
    class="MuiBox-root css-rux501"
  >
    <h1
      class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1vw6mcs-MuiTypography-root"
    >
      Example of Mock Server
    </h1>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
      data-testid="sign-in"
      tabindex="0"
      type="button"
    >
      サインイン
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
      data-testid="get-user"
      tabindex="0"
      type="button"
    >
      ユーザー情報取得
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
